metron-analytics/metron-profiler/src/main/java/org/apache/metron/profiler/bolt/ProfileBuilderBolt.java

```diff
@@ -243,8 +244,18 @@ private void flush(Tuple tickTuple) {
     measurement.setValue(result);
     emit(measurement, tickTuple);
 
+    // Execute the update with the old state
+    Map<String, String> tickUpdate = profileConfig.getTickUpdate();
+    Map<String, Object> state = executor.getState();
+    if(tickUpdate != null) {
+      tickUpdate.forEach((var, expr) -> executor.assign(var, expr, executor.getState()));
```

Comment:
The code above only covers what you describe in this thread; executing 'tick' expressions.  

You still need additional logic to persist the 'tick' state beyond the flush.  And the way that you are doing that in the current code looks fine to me.